### ISSUE TITLE
Implement

### ISSUE THREAD

**Issue by kentcdodds** [formly-js/angular-formly-templates-material#1]
This is just a shell repository @ShMcK is working on an implementation :D

**Comment by kentcdodds** [comment]
@ShMcK, I just noticed the update in the README. Could you give me a link to the issue you posted on angular-material?

**Comment by ShMcK** [comment]
Sure, [posting](https://github.com/angular/material/issues/1478). The issue was closed and I didn't pursue it further.

**Comment by rictorres** [comment]
As usual, they want people to use Google Forums to report/discuss an issue :(
It's like we're back in the 90's, I think we might as well use BBS...

**Comment by Szeliga** [comment]
With the latest releases of angular-material this is possible I think, I've created a basic wrapper and input and everything seems to work.

Wrapper:
```jade
md-input-container
  label {{::to.label}}
  formly-transclude
  div(ng-messages="fc.$error" ng-show="fc.$dirty"
      ng-messages-include="shared/templates/error_messages.jade")
```

Input:
```jade
input(type='{{::to.inputType}}' name='{{::options.key}}' ng-model="model[options.key]")
```

Module run:
```javascript
```

**Comment by kentcdodds** [comment]
Yeah, I actually added `elementAttributes` to angular-formly to support the `flex` and `layout` attributes that angular-material requires. Then I started to implement this library again but got side-tracked. If someone wants to pick it up, that would be amazing :-)

**Comment by kentcdodds** [comment]
A few things this library needs to be considered "done"

- [ ] Update all the types to not be bootstrap-like (some still use bootstrap classes because this was copied from the bootstrap version).
- [ ] Make sure the [mdInputContainerFlex](https://github.com/formly-js/angular-formly-templates-material/blob/master/src/run/mdInputContainerFlex.js) manipulator works
- [ ] Fix the wrappers (still bootstrap stuff here).
- [ ] Update the local demo
- [ ] Add an example that utilizes each type to [the website](http://angular-formly.com)
- [ ] Publish to bower and npm as 1.0.0

Honestly, whoever decides to implement this has as much freedom as they like. I personally have no use case for this right now (which is why it hasn't gotten much love). So whoever builds it can pretty much build it to their own specifications.

**Comment by mmmoli** [comment]
I'm keen to help out.

I'll do what I can.

**Comment by kentcdodds** [comment]
Fantastic! Let me know how I can help you :-)

**Comment by jmls** [comment]
@mmmoli : we are looking to work on this as well - do you have anything to share ?

**Comment by mmmoli** [comment]
@jmls Not really; only that tests wouldn't run.

I'll try to have another crack at it this weekend.
